utils/fuzzer.ts

```diff
@@ -0,0 +1,298 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// jscs:disable
+// jshint ignore: start
+
+'use strict';
+var indexOf = require('lodash.indexof');
+var without = require('lodash.without');
+var maxBy = require('lodash.maxby');
+var random = require('lodash.random');
+var is1 = require('is');
+var isNumber = is1.number;
+var isString = is1.string;
+var isArray = is1.array;
+var isFunction = is1.function;
+
+
+export class Fuzzer {
+  generate = {
+    types: function() {
+      return [
+        'object',
+        'array',
+        'string',
+        'number',
+        'null',
+        'undefined',
+        'function',
+        'boolean',
+      ];
+    },
+
+    string: function(len) {
+      var lenChecked = isNumber(len) ? len : 10;
+      var chars: string[] = [];
+
+      for (var i = 0; i < lenChecked; i++) {
+        chars.push(String.fromCharCode(random(32, 126)));
+      }
+
+      return chars.join('');
+    },
+
+    boolean: function() {
+      return !!random(0, 1);
+    },
+
+    alphaNumericString: function(len) {
+      var lenChecked = isNumber(len) ? len : 10;
+      var chars: string[] = [];
+      var thisRange: number[] = [];
+      var ranges = [[48, 57], [65, 90], [97, 122]];
+
+      for (var i = 0; i < lenChecked; i++) {
+        thisRange = ranges[random(0, 2)];
+        chars.push(String.fromCharCode(random(thisRange[0], thisRange[1])));
+      }
+
+      return chars.join('');
+    },
+
+    function: function() {
+      var availableTypes = without(this.types(), 'function');
+      var typeToGen = this.types()[random(0, availableTypes.length - 1)];
+      var fnToCall = this[typeToGen];
+
+      return function() {
+        return fnToCall();
+      };
+    },
+
+    number: function(lower, upper) {
+      var lowerChecked = isNumber(lower) ? lower : 0;
+      var upperChecked = isNumber(upper) ? upper : 100;
+
+      return random(lowerChecked, upperChecked);
+    },
+
+    null: function() {
+      return null;
+    },
+
+    undefined: function() {
+      return undefined;
+    },
+
+    array: function(
+      len,
+      ofOneType,
+      currentDepth,
+      allowedDepth
+    ) {
+      var lenChecked = isNumber(len) ? len : random(1, 10);
+      var availableTypes =
+        isString(ofOneType) && indexOf(this.types(), ofOneType) > -1
+          ? [ofOneType]
+          : this.types();
+      var currentDepthChecked = isNumber(currentDepth) ? currentDepth : 0;
+      var allowedDepthChecked = isNumber(allowedDepth) ? allowedDepth : 3;
+      var arr: {}[] = [];
```

Comment:
Nit: Prefer Array<{}> here. We prefer Array\<T\> unless the type is just alphanumeric + dot.

Reply:
Fixed